Sources/NIO/MarkedCircularBuffer.swift

```diff
@@ -63,7 +67,7 @@ public struct MarkedCircularBuffer<E>: CustomStringConvertible, AppendableCollec
     }
 
     /// Retrieves the element at the given index from the buffer, without removing it.
-    public subscript(index: Int) -> E {
+    public subscript(index: CircularBuffer<E>.Index) -> E {
```

Comment:
I think we should use a typealias for this type, given that we have to reproduce the generic type name everywhere.

Reply:
good point, thanks! done